addons/sale_project/tests/test_sale_project.py

```diff
@@ -1854,3 +1855,52 @@ def test_enable_milestones_settings_of_project_on_so_confirmation(self):
             so.project_ids.allow_milestones,
             'The generated project should have the "Allow Milestones" setting enabled, as one of the products has invoice policy based on milestones.',
         )
+
+    def test_sale_order_lines_associated_sale_orders_of_parent_and_child_partners(self):
+        """
+        Test Case Steps:
+        1. Create a child partner linked to the parent partner.
+        2. Create sale orders for both the parent and child partners.
+        3. Confirm both sale orders.
+        4. Set the parent partner on the project and fetch the sale order lines.
+        5. Verify that child partner's sale order lines are returned.
+        6. Set the child partner on the project and fetch the sale order lines.
+        7. Verify parent partner's sale order lines are returned.
+        """
+        child_partner = self.env['res.partner'].create({
+            'name': 'Child Partner',
+            'parent_id': self.partner.id,
+        })
+        sale_orders = sale_order_1, sale_order_2 = self.env['sale.order'].create([
+            {
+                'partner_id': self.partner.id,
+                'order_line': [
+                    Command.create({
+                        'product_id': self.product_order_service1.id,
+                        'product_uom_qty': 10,
+                    })
+                ]
+            },
+            {
+                'partner_id': child_partner.id,
+                'order_line': [
+                    Command.create({
+                        'product_id': self.product_order_service2.id,
+                        'product_uom_qty': 5,
+                    })
+                ]
+            }
+        ])
+        sale_orders.action_confirm()
+
+        self.project_global.partner_id = self.partner.id
+        domain = self.project_global._domain_sale_line_id()
+        evaluated_domain = safe_eval(str(domain), {'partner_id': self.project_global.partner_id.id})
```

Comment:
Coucou @odoo/rd-security 

Can we have an override?

We use safe_eval inside a test to correctly eval the domain (since we need to define `partner_id` in the right part of a tuple inside that domain) and check we get the expected SOLs thanks to that domain.

every parameter given to safe_eval function is safe.

Thanks in advance!

Have a nice day!

Reply:
ping @odoo/rd-security 🙂